### ISSUE TITLE
[ENHANCEMENT] Structured logging for Harvester components

### ISSUE THREAD

**Issue by connorkuehl** [harvester/harvester#5402]
**Is your enhancement related to a problem? Please describe.**

There are a number of benefits to using structured logging in Harvester. Firstly, it would make Harvester support bundles more compatible with being fed into something like Elasticsearch (or any alternative). This would allow us to write queries, for example, to see all of the log messages emitted about a specific virtualmachine across all Harvester components.

Ideally, we would be able to feed an entire support bundle into Elasticsearch (or any alternative) to improve our diagnostic capability and establish root causes sooner.

This would improve many more things beyond diagnostic efforts. For example, this could open up more possibilities for Harvester users to customize monitoring/alerting based on queryable logs.

Most of the Harvester codebases already use a structured logging package called "logrus", but we are not taking full advantage of it.

For example, it's common to see logging calls like this in Harvester code:

```go
logrus.Infof("could not shut down %s/%s", vmi.Namespace, vmi.Name)
```

Which would result in a log message looking something like this:

```json
{"msg":"could not shut down default/test"}
```

This is very limiting, as we could only potentially query on strings or substrings in that message, for example:

```
msg == "could not shut down*"
```

Or, if we had the ability to write a query and we wanted to see everything about a specific VM:

```
msg == "test"
```

It was perhaps unfair of me to choose "test" as the VM's name, but there are a number of common words that might conflict with a VM's name, and a query like that would show us a lot of irrelevant logs.

**Describe the solution you'd like**

Instead, Harvester components should standardize on a cross-repo structured logging methodology. Adapting the above example, we would log like this:

```go
logrus.WithFields(logrus.Fields {
    "k8s_namespace": vmi.Namespace,
    "virtualmachine_name": vmi.Name,
}).Info("could not shut down VM")
```

This is a lot more flexible for writing queries against. For example, if we achieved a high level of consistency with logging and how we name our structured log keys, we could search for everything that happened to a VM:

```
virtualmachine_name == "test" AND k8s_namespace == "default"
```

Or query to see all of the VMs we failed to shutdown:

```
msg == "could not shut down VM"
```

**Describe alternatives you've considered**

**Additional context**

**Comment by votdev** [comment]
Can we make a list of field names we want to use? E.g. i do not like the `k8s_namespace`, i suggest to use `namespace` instead.
The field `virtualmachine_name` is used for VMs, what about nodes (`node_name`), Pods (`pod_name`)? Shall we use `<RESOURCE>_name` for example?

**Comment by connorkuehl** [comment]
> Can we make a list of field names we want to use? E.g. i do not like the `k8s_namespace`, i suggest to use `namespace` instead. The field `virtualmachine_name` is used for VMs, what about nodes (`node_name`), Pods (`pod_name`)? Shall we use `<RESOURCE>_name` for example?

Yes, of course. I was just using those for the sake of example. The HEP will be more exhaustive and will prescribe an initial set of keys that we'll all (hopefully) reach consensus on.
